tests/c/phi3.newcg.c

```diff
@@ -20,7 +20,7 @@
 //       %{{15_2}}: i32 = call g() ...
 //     ..~
 //     bb{{_}}:
-//       %{{_}}: i32 = phi bb{{_}} -> %{{15_2}}, bb{{_}} -> %{{13_2}}
+//       %{{_}}: i32 = phi bb{{_}} -> %{{13_2}}, bb{{_}} -> %{{15_2}}
```

Comment:
This test has changed more than is first obvious: notice that line 20 previously referenced `15_2`, but now it's referencing `13_2` (both of these are defined earlier in the test). Do we know why?

Reply:
Looks like mem2reg when iterating over basic blocks and since basic block 13 comes before 15 it renames the phi statement. The test case is failing because we are also checking the value of `%15_2` and `%13_2` which have different values, so the test expects same value in the phi statement.

Reply:
This is an observation but not an explanation: notice that `13_2` is the result of calling `f` but `15_2` is the result of calling `g`. We need a more considered explanation of why this change is OK. What we don't want to do is introduce an optimisation that works for our test cases just by accident. You will probably need to look at the full trace IR for this example and then use that to give an explanation.

Reply:
Phi nodes select the right value based on which predecessor block the control flow came from, regardless of the order. Since mem2reg adds/reconstructs phi nodes it changes the order in which it sees the basic blocks. But this should not be a problem for correctness as the order in phi node doesn't matter anyways.

Reply:
OK that makes sense.